linalg/ode.hpp

```diff
@@ -232,6 +232,8 @@ class AdamsBashforthSolver : public ODESolver
    Vector *k;
    Array<int> idx;
    ODESolver *RKsolver;
+   double dt_;
+   bool print;
```

Comment:
Instead of adding a member variable, can all conditionals that depend on `print` use `mfem::Mpi::Root()` instead?  

From a design point of view, I think it's better to leave the concept of which rank is root encapsulated within the `Mpi` singleton. While I don't think this is really a use case quite yet, such an approach would avoid an issue if the communicator is modified in such a way that root is changed between the construction of the `AdamsBashforthSolver` and the calls to `Step`

Reply:
Noting the [conversation above](https://github.com/mfem/mfem/pull/3466#discussion_r1115963315) regarding calling `Mpi::Root()` from serial examples, I would suggest here moving the conditional logic into a private member function instead of store the value in a member variable:
```c++
inline bool print()
{
#ifdef MFEM_USE_MPI
   return Mpi::IsInitialized() ? Mpi::Root() : true;
#else
  return true;
#endif
}

Reply:
I have made the change but put it in PR #3480.

Reply:
Please excuse my confusion: what is the motivation for putting the change in PR #3480 instead of this one?

Reply:
Noting here that this conversation has been continued in the comments.

Reply:
This PR was meant to fix the double dt issue.
The other PR is a refactorisation and small feature addition. I expected this to be small PR. The dt check was already edit per review request. Perhaps remove it here and really make it a dt fix PR?

Reply:
Done.